pydantic2ts/cli/script.py

```diff
@@ -99,6 +118,25 @@ def extract_pydantic_models(module: ModuleType) -> List[Type[BaseModel]]:
     return models
 
 
+def extract_enum_models(module: ModuleType) -> List[Type[Enum]]:
+    """
+    Given a module, return a list of the Enum classes contained within it.
+    """
+    enums = []
+    module_name = module.__name__
+
+    for _, enum in inspect.getmembers(module, is_enum):
+        if enum.__module__ != "enum":
+            enums.append(enum)
+
+    for _, submodule in inspect.getmembers(
+        module, lambda obj: is_submodule(obj, module_name)
+    ):
+        enums.extend(extract_enum_models(submodule))
```

Comment:
I noticed there's an edge case that this logic for walking module imports isn't able to catch: when there's an import statement of the form `from module import X`, `module` doesn't get listed as a member of the importing module by `inspect.getmembers`. 

So for example:

```py
from another_module import ModelWithEnumField
```
When a field in `ModelWithEnumField` references an enum class defined in `another_module`, the enum class doesn't get extracted by this function, because it won't recurse into `another_module`.

Branch with a test script that demonstrates this: https://github.com/ento/pydantic-to-typescript2/blob/testing-enums/abs_vs_rel.sh

<details><summary>Output from the script</summary>
<p>

```
### Relative import
+ git checkout 972fc5e7e311a43ab14a97e1bfb6e40ca3c39d3d
+ pydantic2ts --module tests/expected_results/enums/v1/input.py --output output.ts --json2ts ./node_modules/.bin/json2ts
2024-10-05 13:27:43,533 Finding pydantic models...
2024-10-05 13:27:43,537 Generating JSON schema from pydantic models...  
cf5cdb5a30c04533858cee0ef94cf0d6                                                       
['AnimalShelter', 'BaseModel', 'Cat', 'Dog', 'Enum', 'List', 'ModelOne', 'ModelTwo', 'Optional', 'Standalone', '__builtins__', '__cached__', '__doc__', '__file__', '__loader_
_', '__name__', '__package__', '__path__', '__spec__', 'schemas']                                                                                                             
cf5cdb5a30c04533858cee0ef94cf0d6.schemas
['__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__', 'schema_one', 'schema_two']
cf5cdb5a30c04533858cee0ef94cf0d6.schemas.schema_one      
['BaseModel', 'Foo', 'ModelOne', '__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__spec__', 'enum']
cf5cdb5a30c04533858cee0ef94cf0d6.schemas.schema_two
['BaseModel', 'Foo', 'ModelTwo', '__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__spec__', 'enum']
2024-10-05 13:27:43,539 Converting JSON schema to typescript definitions...
2024-10-05 13:27:43,731 Saved typescript definitions to output.ts.
+ grep -E 'Foo.? ' output.ts                                                           
export const enum Foo {                                                                
export const enum Foo1 {
+ set +x                                                                               
                                                                                       
### Absolute import
+ git checkout f828b0393d123fbf82b3f45489d61cf81257471d
++ pwd                                                                                 
+ PYTHONPATH=/home/ento/workspace/pydantic-to-typescript2/tests/expected_results/enums/v1
+ pydantic2ts --module /home/ento/workspace/pydantic-to-typescript2/tests/expected_results/enums/v1/input.py --output output.ts --json2ts ./node_modules/.bin/json2ts
2024-10-05 13:27:43,848 Finding pydantic models...
2024-10-05 13:27:43,851 Generating JSON schema from pydantic models...
e270b5d4c08f470fb81b727e49611131
['AnimalShelter', 'BaseModel', 'Cat', 'Dog', 'Enum', 'List', 'ModelOne', 'ModelTwo', 'Optional', 'Standalone', '__builtins__', '__cached__', '__doc__', '__file__', '__loader_
_', '__name__', '__package__', '__path__', '__spec__']
2024-10-05 13:27:43,853 Converting JSON schema to typescript definitions...
2024-10-05 13:27:44,090 Saved typescript definitions to output.ts.
+ grep -E 'Foo.? ' output.ts
export type Foo = "one_a" | "one_b";
export type Foo1 = "two_a" | "two_b";
```

</p>
</details> 

I wonder if going through the list of Pydantic models extracted and recursing into their fields would work instead

Reply:
Good catch! I'll try to find a fix for that!